cura/CuraActions.py

```diff
@@ -229,9 +239,8 @@ def paste(self) -> None:
             if node.callDecoration("isSliceable"):
                 fixed_nodes.append(node)
         # Add the new nodes to the scene, and arrange them
-        group_operation, not_fit_count = createGroupOperationForArrange(nodes, application.getBuildVolume(),
-                                                                        fixed_nodes, factor=10000,
-                                                                        add_new_nodes_in_scene=True)
+        grid_arrange = GridArrange(nodes, application.getBuildVolume(), fixed_nodes)
```

Comment:
Hmm. Not quite sure if I like this approach. So for a "normal" arrange we need to use the createGroupOperationForArrange without a GridArrange object, and for the grid arrange we need to use it with. I think that it would be better to have it more consistent. So either we make an arrange class with a few standard interfaces and with different implementations or we have some kind of factory that handles the behavior.

Reply:
fixed in https://github.com/Ultimaker/Cura/pull/15496/commits/118f49a0520bdd1a3983bbf33b49c3ed8a645fb1

used Nest2Arrange to sort out what kind of arrangement is required. rest all placed replaced it with nest2darrange functions